cartographer/mapping/internal/optimization/optimization_problem_3d.cc

```diff
@@ -421,8 +421,9 @@ void OptimizationProblem3D::Solve(
               result_center_to_center.delta_velocity;
           problem.AddResidualBlock(
               AccelerationCostFunction3D::CreateAutoDiffCostFunction(
-                  options_.acceleration_weight(), delta_velocity,
-                  common::ToSeconds(first_duration),
+                  options_.acceleration_weight() /
+                      common::ToSeconds(first_duration + second_duration),
```

Comment:
I assume it's guaranteed that this can't be zero by some CHECKs before we reach this?

Reply:
I don't know, but this is not a new assumption that nodes do not have the same timestamp, i.e. it would already not work. If you are concerned that we do not CHECK early enough on bad data, I think that should be a separate PR.